### PR TITLE
Use `sample_*` variables if present to populate plink family fields when writing

### LINKED ISSUE #1017: Allow user to specify coordinate as phenotype in `write_plink`
Issue on top of PR https://github.com/pystatgen/sgkit/pull/1003
 
The idea is to give the user an option to specify a custom coordinate to populate the `pheno` property of `to_bed` with. This would make it much easier to write out to plink and directly feed into plink-based GWAS tools.

In terms of spec, one would want to check (at least) that the given coordinate is of dimension `samples` and of a numerical dtype.

### PR DESCRIPTION
**Pull request by tomwhite**
- [x] Fixes #1017 and a part of #1010
- [x] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `changelog.rst`

The idea here is that you can use well-known variables, like `sample_phenotype` to set fields in the plink fam file. This should address #1017.

This isn't finished yet as it needs some tests. Also, I noticed that currently we only support case/control phenotypes (coded as 2 and 1, respectively) and not more general quantative trait values encoded as a float. This would need fixing too - I'm not sure the best way of doing that yet.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
@timothymillar would you mind taking a look through this please? I know very little about plink.

**Comment by timothymillar** [comment]
> I know very little about plink

That makes two of us 😄.

File: sgkit/io/plink/plink_reader.py [hunk 1]
```diff
@@ -106,8 +106,10 @@ def coerce_code(v: dd.Series, codes: List[int]) -> dd.Series:
         v = v.where(v.isin(codes), np.nan)
         return v.fillna(-1).astype("int8")
 
-    df["paternal_id"] = df["paternal_id"].where(df["paternal_id"] != "0", None)
-    df["maternal_id"] = df["maternal_id"].where(df["maternal_id"] != "0", None)
+    # replace fam "0" with sgkit missing values (".")
```
Comment (timothymillar) [inline sgkit/io/plink/plink_reader.py:109]: I think the missing value needs to be documented in the docstring of `read_plink`.

File: sgkit/io/plink/plink_writer.py [hunk 2]
```diff
@@ -20,6 +31,20 @@ def write_plink(
 ) -> None:
```
Comment (timothymillar) [inline sgkit/io/plink/plink_writer.py:31]: Should this have a `missing: Hashable ='.' ` parameter? To specify the missing parent value and default to the new value from `read_plink`.

File: sgkit/io/plink/plink_writer.py [hunk 3]
```diff
@@ -78,11 +103,12 @@ def write_plink(
...
         "allele_2": ds.variant_allele.values[:, 1],
-        "iid": ds.sample_id,
+        "iid": ds.sample_id.values,  # may be overridden by sample_member_id below (if present)
     }
 
-    if "variant_id" in ds:
-        properties["sid"] = ds.variant_id
+    for var, prop in FAM_VARIABLE_TO_BED_READER.items():
+        if var in ds:
+            properties[prop] = ds[var].values
```
Comment (timothymillar) [inline sgkit/io/plink/plink_writer.py:111]: I might be missing something but does the new/optional missing value `'.'` need to be transformed back to `0`?

File: sgkit/io/plink/plink_writer.py (hunk 2 above)
Comment (tomwhite) [reply sgkit/io/plink/plink_writer.py:31]: I'm not sure this is needed. We don't parameterise the missing value in VCF for example, so I think it's ok to have the sgkit default of "." used here too.

File: sgkit/io/plink/plink_writer.py (hunk 3 above)
Comment (tomwhite) [reply sgkit/io/plink/plink_writer.py:111]: Done

**Comment by tomwhite** [comment]
Thanks for the review @timothymillar. I've updated the PR with changes.

File: sgkit/io/plink/plink_writer.py (hunk 2 above)
Comment (timothymillar) [reply sgkit/io/plink/plink_writer.py:33]: Ah, I see it's handled by `values = np.where(values == STR_MISSING, "0", values)`. 
> We don't parameterise the missing value in VCF

Good point, but "." is the missing value in the VCF spec where as "0" is the missing value in a plink fam file. Although, a "0" will stay as a "0" so I guess both cases are covered anyway 🤷

> the sgkit default of "."

We should note that in the main documentation (not needed in this PR)

**Review by timothymillar — APPROVED** [review]
LGTM!

**Pull request merged**